cli/tests/pcluster/config/test_update_policy.py

```diff
@@ -1295,3 +1298,110 @@ def test_condition_checker_managed_placement_group(
     assert_that(actual_message).is_equal_to(expected_message)
+
+
+@pytest.mark.parametrize(
+    "base_config, target_config, change, expected_subnet_updated, expected_fail_reason, expected_action_needed",
+    [
+        # If change includes SubnetIds and existing + new cluster configuration uses the same managed Fsx for Lustre
+        #   - Show validation failure message
+        (
+            {
+                "Scheduling": {"Queues": [{"Name": "mock-q", "Networking": {"SubnetIds": ["subnet-12345678"]}}]},
+                "SharedStorage": [
+                    {"MountDir": "/test-fsx-lustre", "Name": "test-fsx-lustre", "StorageType": "FsxLustre"}
+                ],
+            },
+            {
+                "Scheduling": {"Queues": [{"Name": "mock-q", "Networking": {"SubnetIds": ["subnet-87654321"]}}]},
+                "SharedStorage": [
+                    {"MountDir": "/test-fsx-lustre", "Name": "test-fsx-lustre", "StorageType": "FsxLustre"}
+                ],
+            },
+            Change(
+                path=["Queues[mock-q]", "Networking"],
+                key="SubnetIds",
+                old_value=["subnet-12345678"],
+                new_value=["subnet-87654321"],
+                update_policy={},
+                is_list=False,
+            ),
+            False,
+            "Updating the SubnetIds parameter will cause these FSx for Lustre file system(s) to be replaced: "
+            "{'test-fsx-lustre'}",
+            "If you intend to proceed with the update, please make sure to back-up your data and explicitly replace "
+            "the file system(s) ({'test-fsx-lustre'}) with a new one(s) in the cluster configuration.",
+        ),
+        # If update includes SubnetIds and existing cluster configuration uses an External Fsx for Lustre FS
+        #   - Fall back to QueueUpdateStrategy Update Policy failure message
+        (
+            {
+                "Scheduling": {"Queues": [{"Name": "mock-q", "Networking": {"SubnetIds": ["subnet-12345678"]}}]},
+                "SharedStorage": [
+                    {"MountDir": "/test-fsx-lustre", "FileSystemId": "test-fsx-lustre-id", "StorageType": "FsxLustre"}
+                ],
+            },
+            {
+                "Scheduling": {"Queues": [{"Name": "mock-q", "Networking": {"SubnetIds": ["subnet-87654321"]}}]},
+                "SharedStorage": [
+                    {"MountDir": "/test-fsx-lustre", "FileSystemId": "test-fsx-lustre-id", "StorageType": "FsxLustre"}
+                ],
+            },
+            Change(
+                path=["Queues[mock-q]", "Networking"],
+                key="SubnetIds",
+                old_value=["subnet-12345678"],
+                new_value=["subnet-87654321"],
+                update_policy={},
+                is_list=False,
+            ),
+            False,
+            "All compute nodes must be stopped",
+            "Stop the compute fleet with the pcluster update-compute-fleet command",
```

Comment:
it should be
"Stop the compute fleet with the pcluster update-compute-fleet command, "
"or set QueueUpdateStrategy in the configuration used for the 'update-cluster' operation",